Assets/Scripts/Enemy/EnemyController.cs

```diff
@@ -34,12 +41,31 @@ public void TakeDamage(int value)
             }
         }
 
+        public void ChangeState(EnemyState enemyState)
+        {
+            lastStateView = currentStateView;
+
+            currentStateView = enemyState;
+
+            if (currentStateView == EnemyState.petrol)
+                enemyView.PetrolState.enabled = true;
```

Comment:
you are not disabling the last state?

Reply:
Doing it in state itself, when it calls to change the state